importlib_resources/tests/test_resource.py

```diff
@@ -186,15 +186,35 @@ def test_submodule_contents(self):
             contents.remove('__pycache__')
         except KeyError:
             pass
-        self.assertEqual(contents, {'binary.file', 'utf-8.file', 'utf-16.file'})
+        self.assertEqual(
+            contents, {'subdirectory', 'binary.file', 'utf-8.file', 'utf-16.file'}
+        )
 
     def test_submodule_contents_by_name(self):
         contents = names(resources.files('namespacedata01'))
         try:
             contents.remove('__pycache__')
         except KeyError:
             pass
-        self.assertEqual(contents, {'binary.file', 'utf-8.file', 'utf-16.file'})
+        self.assertEqual(
+            contents, {'subdirectory', 'binary.file', 'utf-8.file', 'utf-16.file'}
+        )
+
+    def test_submodule_sub_contents(self):
+        contents = names(resources.files(import_module('namespacedata01.subdirectory')))
```

Comment:
Since we're testing a "submodule" here, maybe the directory should be named "submodule"?

Reply:
I started going about renaming 'subdirectory' to 'submodule' (217187c) but I realized that 'subdirectory' is used as a submodule in the data01 tests, so it's probably better to re-use that name (as you've done).